### ISSUE TITLE
Deserialisation crash on null user image

### ISSUE THREAD

**Issue by wang-li** [GetStream/stream-chat-android#3278]
**Describe the bug**
A crash happens whenever a user has a `null` image in the Stream backend.
Here is the stack trace:
```
2022-04-01 16:28:43.159 11368-11666 E/Chat:NEW_SERIALIZATION_ERROR: com.squareup.moshi.JsonDataException: com.squareup.moshi.JsonDataException: Non-null value 'image' was null at $.image at $.channels[25].read[9].user
        at com.squareup.moshi.AdapterMethodsFactory$1.fromJson(AdapterMethodsFactory.java:101)
        at com.squareup.moshi.Moshi$Lookup.fromJson(Moshi.java:409)
        at io.getstream.chat.android.client.api2.model.dto.DownstreamChannelUserReadJsonAdapter.fromJson(DownstreamChannelUserReadJsonAdapter.kt:47)
        at io.getstream.chat.android.client.api2.model.dto.DownstreamChannelUserReadJsonAdapter.fromJson(DownstreamChannelUserReadJsonAdapter.kt:22)
        at com.squareup.moshi.internal.NullSafeJsonAdapter.fromJson(NullSafeJsonAdapter.java:41)
        at com.squareup.moshi.CollectionJsonAdapter.fromJson(CollectionJsonAdapter.java:81)
        at com.squareup.moshi.CollectionJsonAdapter$2.fromJson(CollectionJsonAdapter.java:55)
        at com.squareup.moshi.internal.NullSafeJsonAdapter.fromJson(NullSafeJsonAdapter.java:41)
        at io.getstream.chat.android.client.api2.model.response.ChannelResponseJsonAdapter.fromJson(ChannelResponseJsonAdapter.kt:106)
        at io.getstream.chat.android.client.api2.model.response.ChannelResponseJsonAdapter.fromJson(ChannelResponseJsonAdapter.kt:32)
        at com.squareup.moshi.internal.NullSafeJsonAdapter.fromJson(NullSafeJsonAdapter.java:41)
        at com.squareup.moshi.CollectionJsonAdapter.fromJson(CollectionJsonAdapter.java:81)
        at com.squareup.moshi.CollectionJsonAdapter$2.fromJson(CollectionJsonAdapter.java:55)
        at com.squareup.moshi.internal.NullSafeJsonAdapter.fromJson(NullSafeJsonAdapter.java:41)
        at io.getstream.chat.android.client.api2.model.response.QueryChannelsResponseJsonAdapter.fromJson(QueryChannelsResponseJsonAdapter.kt:39)
        at io.getstream.chat.android.client.api2.model.response.QueryChannelsResponseJsonAdapter.fromJson(QueryChannelsResponseJsonAdapter.kt:22)
        at com.squareup.moshi.internal.NullSafeJsonAdapter.fromJson(NullSafeJsonAdapter.java:41)
        at retrofit2.converter.moshi.MoshiResponseBodyConverter.convert(MoshiResponseBodyConverter.java:46)
        at retrofit2.converter.moshi.MoshiResponseBodyConverter.convert(MoshiResponseBodyConverter.java:27)
        at io.getstream.chat.android.client.parser2.MoshiErrorLoggingKt$withErrorLogging$1.responseBodyConverter$lambda-0(MoshiErrorLogging.kt:25)
        at io.getstream.chat.android.client.parser2.MoshiErrorLoggingKt$withErrorLogging$1.$r8$lambda$avdRyQtCDxTxehfl0DwpRaWHnVs(Unknown Source:0)
        at io.getstream.chat.android.client.parser2.MoshiErrorLoggingKt$withErrorLogging$1$$ExternalSyntheticLambda0.convert(Unknown Source:6)
        at retrofit2.OkHttpCall.parseResponse(OkHttpCall.java:243)
        at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:153)
        at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:920)
     Caused by: com.squareup.moshi.JsonDataException: Non-null value 'image' was null at $.image
        at com.squareup.moshi.internal.Util.unexpectedNull(Util.java:674)
        at io.getstream.chat.android.client.api2.model.dto.DownstreamUserDtoJsonAdapter.fromJson(DownstreamUserDtoJsonAdapter.kt:106)
        at io.getstream.chat.android.client.api2.model.dto.DownstreamUserDtoJsonAdapter.fromJson(DownstreamUserDtoJsonAdapter.kt:29)
        at com.squareup.moshi.internal.NullSafeJsonAdapter.fromJson(NullSafeJsonAdapter.java:41)
        at com.squareup.moshi.JsonAdapter.fromJsonValue(JsonAdapter.java:126)
        at io.getstream.chat.android.client.parser2.adapters.CustomObjectDtoAdapter.parseWithExtraData(CustomObjectDtoAdapter.kt:65)
        at io.getstream.chat.android.client.parser2.adapters.DownstreamUserDtoAdapter.fromJson(UserDtoAdapters.kt:19)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.squareup.moshi.AdapterMethodsFactory$AdapterMethod.invoke(AdapterMethodsFactory.java:383)
        at com.squareup.moshi.AdapterMethodsFactory$4.fromJson(AdapterMethodsFactory.java:269)
2022-04-01 16:28:43.159 11368-11666 E/Chat:NEW_SERIALIZATION_ERROR:     at com.squareup.moshi.AdapterMethodsFactory$1.fromJson(AdapterMethodsFactory.java:97)
        	... 27 more
```

**SDK version**
- 5.0.1

**Expected behavior**
The SDK should not crash if a user has a null image.

**Comment by adasiewiczr** [comment]
Hey @wang-li,
thanks for reporting the issue. It was fixed in the PR mentioned above and will be shipped in the next release!

**Comment by adasiewiczr** [comment]
Hey @wang-li,
we released a new version (5.0.3) that includes the fix mentioned above. You can find more details [here](https://github.com/GetStream/stream-chat-android/releases/tag/v5.0.3).
Don't hesitate to contact us if the issue still exists.
